### PR TITLE
Added summary for mask generation

### PR DESCRIPTION
**Pull request by NebulaAnish**
Added summary for mask generation. It is a small contribution to familiarize with the process. I will contribute more to this section once my first PR gets approved.

### CODE REVIEW

File: packages/tasks/src/tasks/mask-generation/data.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import type { TaskDataCustom } from "..";
+
+const taskData: TaskDataCustom = {
+	datasets: [
+	],
+	demo: {
+		inputs: [
```
Comment (merveenoyan) [inline packages/tasks/src/tasks/mask-generation/data.ts:7]:
```suggestion
		inputs: [],
```

File: packages/tasks/src/tasks/mask-generation/data.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import type { TaskDataCustom } from "..";
+
+const taskData: TaskDataCustom = {
+	datasets: [
```
Comment (merveenoyan) [inline packages/tasks/src/tasks/mask-generation/data.ts:4]:
```suggestion
	datasets: [],
```

File: packages/tasks/src/tasks/mask-generation/data.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+	],
```
Comment (merveenoyan) [inline packages/tasks/src/tasks/mask-generation/data.ts:5]:
```suggestion
```

File: packages/tasks/src/tasks/mask-generation/data.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+	demo: {
+		inputs: [
+
```
Comment (merveenoyan) [inline packages/tasks/src/tasks/mask-generation/data.ts:8]:
```suggestion
```

File: packages/tasks/src/tasks/mask-generation/data.ts [hunk 5]
```diff
@@ -0,0 +1,27 @@
+import type { TaskDataCustom } from "..";
+
+const taskData: TaskDataCustom = {
+	datasets: [
+	],
+	demo: {
+		inputs: [
+			
+		],
```
Comment (merveenoyan) [inline packages/tasks/src/tasks/mask-generation/data.ts:9]:
```suggestion
```

File: packages/tasks/src/tasks/mask-generation/data.ts [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+		outputs: [
```
Comment (merveenoyan) [inline packages/tasks/src/tasks/mask-generation/data.ts:10]:
```suggestion
		outputs: [],
```

File: packages/tasks/src/tasks/mask-generation/data.ts [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+
```
Comment (merveenoyan) [inline packages/tasks/src/tasks/mask-generation/data.ts:11]:
```suggestion
```

File: packages/tasks/src/tasks/mask-generation/data.ts [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+const taskData: TaskDataCustom = {
+	datasets: [
+	],
+	demo: {
+		inputs: [
+			
+		],
+		outputs: [
+			
+		],
```
Comment (merveenoyan) [inline packages/tasks/src/tasks/mask-generation/data.ts:12]:
```suggestion
```

File: packages/tasks/src/tasks/mask-generation/data.ts [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+	},
+	metrics: [],
+	models: [
+
```
Comment (merveenoyan) [inline packages/tasks/src/tasks/mask-generation/data.ts:16]:
```suggestion
```

File: packages/tasks/src/tasks/mask-generation/data.ts [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+	demo: {
+		inputs: [
+			
+		],
+		outputs: [
+			
+		],
+	},
+	metrics: [],
+	models: [
```
Comment (merveenoyan) [inline packages/tasks/src/tasks/mask-generation/data.ts:15]:
```suggestion
	models: [],
```

File: packages/tasks/src/tasks/mask-generation/data.ts [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+		
+	],
```
Comment (merveenoyan) [inline packages/tasks/src/tasks/mask-generation/data.ts:17]:
```suggestion
```

File: packages/tasks/src/tasks/mask-generation/data.ts [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+	spaces: [
```
Comment (merveenoyan) [inline packages/tasks/src/tasks/mask-generation/data.ts:18]:
```suggestion
	spaces: [],
```

File: packages/tasks/src/tasks/mask-generation/data.ts [hunk 13]
```diff
@@ -0,0 +1,27 @@
...
+
```
Comment (merveenoyan) [inline packages/tasks/src/tasks/mask-generation/data.ts:19]:
```suggestion
```

File: packages/tasks/src/tasks/mask-generation/data.ts [hunk 14]
```diff
@@ -0,0 +1,27 @@
...
+			
+		],
+	},
+	metrics: [],
+	models: [
+		
+	],
+	spaces: [
+		
+	],
```
Comment (merveenoyan) [inline packages/tasks/src/tasks/mask-generation/data.ts:20]:
```suggestion
```

File: packages/tasks/src/tasks/mask-generation/data.ts [hunk 15]
```diff
@@ -0,0 +1,27 @@
...
+	summary:
+		"Mask generation refers to the process of creating a binary image that identifies a specific object or region of interest in an input image. Masks are often used in object detection and segmentation tasks, where they provide a precise way to isolate the object of interest for further processing or analysis.",
+	widgetModels: [""],
```
Comment (merveenoyan) [inline packages/tasks/src/tasks/mask-generation/data.ts:23]:
```suggestion
	widgetModels: [],
```

File: packages/tasks/src/tasks/mask-generation/data.ts [hunk 16]
```diff
@@ -0,0 +1,27 @@
...
+	},
+	metrics: [],
+	models: [
+		
+	],
+	spaces: [
+		
+	],
+	summary:
+		"Mask generation refers to the process of creating a binary image that identifies a specific object or region of interest in an input image. Masks are often used in object detection and segmentation tasks, where they provide a precise way to isolate the object of interest for further processing or analysis.",
```
Comment (merveenoyan) [inline packages/tasks/src/tasks/mask-generation/data.ts:22]:
```suggestion
		"Mask generation is creating a binary image that identifies a specific object or region of interest in an input image. Masks are often used in segmentation tasks, where they provide a precise way to isolate the object of interest for further processing or analysis.",
```

**Review by merveenoyan — COMMENTED** [review]
Thank you! just added a bunch of nits to format

**Comment by NebulaAnish** [comment]
Thanks for this. What should I do next to get this PR merged. I'm new to open source.

**Comment by NebulaAnish** [comment]
Thank you for the resource. I am looking to contribute more to this section. I'll make sure to give contribution guide a thorough read as well.

**Comment by merveenoyan** [comment]
@NebulaAnish thank you, I just ran CI and realized the lint check was failing. I will open a PR to your branch to fix linting later and we can merge.

**Comment by merveenoyan** [comment]
@coyotte508 I pulled the author's repository and ran formatting (without check) however after formatting and linting realized there's nothing to commit. Do you know what might be wrong here?

**Comment by NebulaAnish** [comment]
Thank you for merging 🫶

**Pull request merged**